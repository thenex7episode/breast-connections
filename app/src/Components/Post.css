.metaContainer {
    text-align: right;
}

.titleContainer {
    padding: 1em;
    display: flex;
    justify-content: space-between;
    border: 1px solid grey;
}

.contentContainer {
    padding: 1em;
    border: 1px solid grey;
    display: flex;
    justify-content: space-between;
}

.metaContainer {
    display: flex;
}

.trackerBadge {
    padding: 0 0.5em;
}