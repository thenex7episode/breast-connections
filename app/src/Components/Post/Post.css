.postContainer {
    margin: 2em 0;
    border: 1px solid #F433AB;
    border-radius: 2px;
}

.titleContainer {
    padding: 0.5em;
    display: flex;
    justify-content: space-between;
}

.postBody {
    border-top: 0;
    padding: 0.5em;
}

.trackerBadge {
    padding: 0 0.5em;
    display: block;
}

.editContainer {
    position: absolute;
    right: 1.5em;
    background: white;
    z-index: 1;
    border: 1px solid #eee;
}

.editContainer > li {
    float: none;
    color: grey;
    font-size: 10pt;
    width: 150px;
}

.contentContainer {
    display: flex;
    justify-content: space-between;
}

.imageList {
    width: 45%;
    height: auto;
    margin: 1em;
}

.postImage {
    max-width: 100%;
    max-height: 100%;
}

.imageContainer {
    width: 50px;
}