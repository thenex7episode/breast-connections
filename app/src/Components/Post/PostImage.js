import React, { Component } from 'react';
import axios from 'axios';

export default class PostImage extends Component {
    constructor(){
        super();
        this.state = {
            images: []
        }
    }

    componentDidMount(){
        axios.get(`/api/images/${this.props.post_id}`).then(response => {
            this.setState({images: response.data})
        })
    }

    render() {
        const { post_id } = this.props;
        const imageList = this.state.images.map((el,i) => <div className='imageList' ><img className='postImage' src={el.url} alt={i}/></div>)
        return (
            <div style={{display: 'flex', flexWrap: 'wrap', justifyContent: 'center'}}>
                {imageList}
            </div>
        );
    }
}