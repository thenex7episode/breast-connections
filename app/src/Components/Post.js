import React, { Component } from 'react';
import axios from 'axios';
import './Post.css';
import { Avatar, Icon, Badge, Button } from 'antd';
import { Menu, Dropdown } from 'antd';
import { relative } from 'path';

export default class Post extends Component {
    constructor(props){
        super(props);
        this.state = {
            loggedUser: '',
            username: '',
            userImage: '',
            tracker: this.props.tracker,
            edit: false,
            userPost: true
        }
    }

    componentDidMount(){
        axios.get(`/api/user/${this.props.user_id}`).then(data => {
            this.setState({username: data.data[0].username, userImage: data.data[0].imageurl})
        })
    }

    increaseTracker(){
        const post = {
            post_id: this.props.post_id,
            tracker: this.state.tracker+1
        }
        axios.put(`/api/editpost/`, post).then(data => {
            console.log(data.data.data[0].tracker)
            this.setState({tracker: data.data.data[0].tracker})
        })
    }

    deletePostFn(){
        this.props.deletePostFn(this.props.post_id)
        this.setState({edit: false})
    }


    render() {
        const { title, body, date, tracker, user_id, deletePostFn } = this.props;
        const { username, userImage } = this.state;
        return (
            <div className='postContainer'>
                <div className='titleContainer'>
                    <div style={{padding: '1em'}}>
                        <div style={{fontSize: '1.5em'}}>{title}</div>
                        <div style={{fontSize: '0.8em'}}>posted on {date}, by {username}</div>
                    </div>
                    <div>
                        <Icon onClick={() => this.setState({edit: !this.state.edit})} style={{display: 'block', opacity: this.state.userPost ? '1' : '0', paddingBottom: '1em'}}type="ellipsis" />
                            <ul style={{display: this.state.edit ? 'block' : 'none'}} className='editContainer'>
                                <li onClick={() => this.deletePostFn()}><Icon type="delete" /> Delete</li>
                                <li><Icon type="edit" /> Edit</li>
                            </ul>
                        <Avatar src={userImage} />
                    </div>
                </div>
                <div className='contentContainer'>
                    <p>{body}</p>
                    <div style={{}}>
                        <Badge className='trackerBadge' count={this.state.tracker} style={{ backgroundColor: 'green' }} />
                        <Button onClick={() => this.increaseTracker()} shape="circle" icon="up" />
                    </div>
                </div>
            </div>
        );
    }
}