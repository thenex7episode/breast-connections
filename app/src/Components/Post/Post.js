import React, { Component } from 'react';
import axios from 'axios';
import './Post.css';
import { Avatar, Icon, Badge, Button, Input, Menu, Dropdown, Spin } from 'antd';
import { relative } from 'path';
import Comments from '../Comments/Comments';
const { TextArea } = Input;

const antIcon = <Icon type="loading" style={{ fontSize: 24 }} spin />;


export default class Post extends Component {
    constructor(props){
        super(props);
        this.state = {
            loggedUser: '',
            username: '',
            userImage: '',
            tracker: this.props.tracker,
            edit: false,
            editMode: false,
            titleInput: null,
            bodyInput: null,
            userPost: true,
            collapsed: false,
            comments: null
        }
    }

    componentDidMount(){
        axios.get(`/api/posts/${this.props.user_id}`).then(data => {
            console.log(data)
            this.setState({username: data.data[0].username, userImage: data.data[0].imageurl})
        })
    }

    increaseTracker(){
        const post = {
            post_id: this.props.post_id,
            tracker: this.state.tracker+1
        }
        axios.put(`/api/addtracker/`, post).then(data => {
            console.log(data.data.data[0].tracker)
            this.setState({tracker: data.data.data[0].tracker})
        })
    }

    deletePostFn(){
        this.props.deletePostFn(this.props.post_id)
        this.setState({edit: false})
    }

    editPost(body, title){
        const post = {
            post_id: this.props.post_id,
            title: this.state.titleInput || title,
            body: this.state.bodyInput || body
        }
        axios.put('/api/editpost/', post).then(data => {
            this.setState({editMode: false})
        })
    }


    render() {
        const { title, body, date, tracker, user_id, deletePostFn, loggedUser, post_id } = this.props;
        const { username, userImage } = this.state;
        <Comments loggedUser={loggedUser} post_id={post_id}/>
        return (
            <div className='postContainer'>
                <div className='titleContainer'>
                    <div style={{padding: '1em'}}>
                        <Input style={{fontSize: '1.5em', display: this.state.editMode ? 'block' : 'none'}} value={this.state.titleInput === null ? title : this.state.titleInput} onChange={e => this.setState({titleInput: e.target.value})}/>
                        <div style={{fontSize: '1.5em', display: this.state.editMode ? 'none' : 'block'}}>{this.state.titleInput || title}</div>
                        <div style={{fontSize: '0.8em'}}>posted on {date}, by {username}</div>
                    </div>
                    <div>
                        <Icon onClick={loggedUser === this.state.username ? () => this.setState({edit: !this.state.edit}): ''} style={{display: 'block' ,opacity: loggedUser === this.state.username ? '1':'0', paddingBottom: '1em'}}type="ellipsis" />
                            <ul style={{display: this.state.edit ? 'block' : 'none'}} className='editContainer'>
                                <li onClick={() => this.deletePostFn()}><Icon type="delete" /> Delete</li>
                                <li onClick={() => this.setState({editMode: !this.state.editMode, edit: false})}><Icon type="edit" /> Edit</li>
                            </ul>
                        <Avatar src={userImage} />
                    </div>
                </div>
                <div className='postBody'>
                    <div className='contentContainer'>
                        <TextArea style={{fontSize: '1.5em', display: this.state.editMode ? 'block' : 'none'}} value={this.state.bodyInput === null ? body : this.state.bodyInput} onChange={e => this.setState({bodyInput: e.target.value})}/>
                        <Button style={{display: this.state.editMode ? 'block' : 'none'}} onClick={() => this.editPost(body, title)} >Save</Button>
                        <p style={{padding: '1em', display: this.state.editMode ? 'none' : 'block'}}>{this.state.bodyInput || body}</p>
                        <div>
                            <Badge className='trackerBadge' count={this.state.tracker} style={{ backgroundColor: 'green' }} />
                            <Button onClick={() => this.increaseTracker()} shape="circle" icon="up" />
                        </div>
                    </div>
                    <div style={{textAlign: 'center'}}>
                        <div style={{display: this.state.collapsed ? 'block' : 'none'}}>
                            {this.state.comments}
                        </div>
                        <Button onClick={() => this.setState({collapsed: !this.state.collapsed, comments: <Comments loggedUser={loggedUser} post_id={post_id}/>})}>Read</Button>
                    </div>
                </div>
            </div>
        );
    }
}