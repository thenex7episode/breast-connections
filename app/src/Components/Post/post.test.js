import Post from './Post'

it("renders correctly", () => {
    const wrapper = shallow(
        <Post />
    );
    expect(wrapper).toMatchSnapshot()
});

