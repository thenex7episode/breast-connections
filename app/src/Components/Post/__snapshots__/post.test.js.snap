// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="postContainer"
>
  <div
    className="titleContainer"
  >
    <div
      style={
        Object {
          "padding": "1em",
        }
      }
    >
      <Input
        disabled={false}
        onChange={[Function]}
        prefixCls="ant-input"
        style={
          Object {
            "display": "none",
            "fontSize": "1.5em",
          }
        }
        type="text"
      />
      <div
        style={
          Object {
            "display": "block",
            "fontSize": "1.5em",
          }
        }
      />
      <div
        style={
          Object {
            "fontSize": "0.8em",
          }
        }
      >
        posted on 
        , by 
      </div>
    </div>
    <div>
      <Icon
        onClick=""
        style={
          Object {
            "display": "block",
            "opacity": "0",
            "paddingBottom": "1em",
          }
        }
        type="ellipsis"
      />
      <ul
        className="editContainer"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <li
          onClick={[Function]}
        >
          <Icon
            type="delete"
          />
           Delete
        </li>
        <li
          onClick={[Function]}
        >
          <Icon
            type="edit"
          />
           Edit
        </li>
      </ul>
      <Avatar
        prefixCls="ant-avatar"
        shape="circle"
        size="default"
        src=""
      />
    </div>
  </div>
  <div
    className="postBody"
  >
    <div>
      <PostImage />
    </div>
    <div
      className="contentContainer"
    >
      <TextArea
        onChange={[Function]}
        prefixCls="ant-input"
        style={
          Object {
            "display": "none",
            "fontSize": "1.5em",
          }
        }
      />
      <Button
        ghost={false}
        loading={false}
        onClick={[Function]}
        prefixCls="ant-btn"
        style={
          Object {
            "display": "none",
          }
        }
      >
        Save
      </Button>
      <p
        style={
          Object {
            "display": "block",
            "padding": "1em",
          }
        }
      />
      <div>
        <Badge
          className="trackerBadge"
          count={null}
          dot={false}
          overflowCount={99}
          prefixCls="ant-badge"
          scrollNumberPrefixCls="ant-scroll-number"
          showZero={false}
          style={
            Object {
              "backgroundColor": "green",
            }
          }
        />
        <Button
          ghost={false}
          icon="up"
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
          shape="circle"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        style={
          Object {
            "display": "none",
          }
        }
      />
      <Button
        ghost={false}
        loading={false}
        onClick={[Function]}
        prefixCls="ant-btn"
      >
        Read
      </Button>
    </div>
  </div>
</div>
`;
